### PR TITLE
Created tests for cli/add_relationship

### PR DESCRIPTION
**Pull request by patrikgrenfeldt**
This PR adds some tests for cli/add_relationship.

How to test:
1. open PR

Expected outcome:
Travis build succeeds, tests are more numerous than before and all tests pass

### CODE REVIEW

File: cg/cli/add.py [hunk 1]
```diff
@@ -137,18 +137,38 @@ def family(context, priority, panels, customer_id, name):
 @add.command()
 @click.option('-m', '--mother', help='sample ID for mother of sample')
 @click.option('-f', '--father', help='sample ID for father of sample')
-@click.option('-s', '--status', type=click.Choice(['affected', 'unaffected', 'unknown']),
+@click.option('-s', '--status', type=click.Choice(STATUS_OPTIONS),
```
Comment (ingkebil) [inline cg/cli/add.py]: This is cool :)

File: cg/store/models.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 import alchy
 from cg.constants import REV_PRIORITY_MAP, PRIORITY_MAP, FAMILY_ACTIONS, FLOWCELL_STATUS, \
-    PREP_CATEGORIES
+    PREP_CATEGORIES, STATUS_OPTIONS
```
Comment (ingkebil) [inline cg/store/models.py]: Why are the STATUS_OPTIONS here?

File: tests/cli/test_cli_add_relationship.py [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+        ['--database', db_uri, 'add', 'relationship', family_id,
+         sample_id, '-s', status])
+
+    # THEN then it should complain on bad status instead of adding a relationship
+    assert result.exit_code == 2
+    assert disk_store.FamilySample.query.count() == 0
+
+
+def test_add_relationship_mother(invoke_cli, disk_store: Store):
+    """Test to add a relationship using only the required arguments"""
```
Comment (ingkebil) [inline tests/cli/test_cli_add_relationship.py]:
```suggestion
    """Test to add a relationship with a mother"""
```

File: tests/cli/test_cli_add_relationship.py [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+         family_id, sample_id, '-s', status, '-m', mother_id])
+
+    # THEN then it should be added
+    assert result.exit_code == 0
+    assert disk_store.FamilySample.query.count() == 1
+    assert disk_store.FamilySample.query.first().mother.internal_id == mother_id
+
+
+def test_add_relationship_bad_mother(invoke_cli, disk_store: Store):
+    """Test to add a relationship using only the required arguments"""
```
Comment (ingkebil) [inline tests/cli/test_cli_add_relationship.py]:
```suggestion
    """Test to add a relationship using a non-existing mother"""
```

File: tests/cli/test_cli_add_relationship.py [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+        ['--database', db_uri, 'add', 'relationship',
+         family_id, sample_id, '-s', status, '-m', mother_id])
+
+    # THEN then it should not be added
+    assert result.exit_code == 1
+    assert disk_store.FamilySample.query.count() == 0
+
+
+def test_add_relationship_father(invoke_cli, disk_store: Store):
+    """Test to add a relationship using only the required arguments"""
```
Comment (ingkebil) [inline tests/cli/test_cli_add_relationship.py]:
```suggestion
    """Test to add a relationship using a father"""
```

File: tests/cli/test_cli_add_relationship.py [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+    # GIVEN a database with a sample and an family
+    sample_id = add_sample(disk_store)
+    father_id = add_sample(disk_store, 'father')
```
Comment (ingkebil) [inline tests/cli/test_cli_add_relationship.py]: ... you're adding a female father? :D

File: tests/cli/test_cli_add_relationship.py [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+         family_id, sample_id, '-s', status, '-f', father_id])
+
+    # THEN then it should be added
+    assert result.exit_code == 0
+    assert disk_store.FamilySample.query.count() == 1
+    assert disk_store.FamilySample.query.first().father.internal_id == father_id
+
+
+def test_add_relationship_bad_father(invoke_cli, disk_store: Store):
+    """Test to add a relationship using only the required arguments"""
```
Comment (ingkebil) [inline tests/cli/test_cli_add_relationship.py]:
```suggestion
    """Test to add a relationship using a non-existing father"""
```

File: tests/cli/test_cli_add_relationship.py [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+        ['--database', db_uri, 'add', 'relationship',
+         family_id, sample_id, '-s', status, '-f', father_id])
+
+    # THEN then it should not be added
+    assert result.exit_code == 1
+    assert disk_store.FamilySample.query.count() == 0
+
+
+def ensure_application_version(disk_store, application_tag='dummy_tag'):
+    """utility function to return existing or create application version for tests"""
```
Comment (ingkebil) [inline tests/cli/test_cli_add_relationship.py]: Is this still needed? seem unused

File: tests/cli/test_cli_add_relationship.py [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+    if not version:
+        version = disk_store.add_version(application, 1, valid_from=datetime.now(),
+                                         prices=prices)
+
+        disk_store.add_commit(version)
+    return version.id
+
+
+def ensure_customer(disk_store, customer_id='cust_test'):
+    """utility function to return existing or create customer for tests"""
```
Comment (ingkebil) [inline tests/cli/test_cli_add_relationship.py]: Is this function still needed?

File: tests/cli/test_cli_add_relationship.py (hunk 6 above)
Comment (patrikgrenfeldt) [reply tests/cli/test_cli_add_relationship.py]: fixed

**Pull request merged**